apps/common/test/common_test.exs

```diff
@@ -0,0 +1,4 @@
+defmodule CommonTest do
+  use ExUnit.Case
+  doctest Common
```

Comment:
it'd be nice to have test cases for those `valid?` functions

Reply:
yeah, I am on it.
may be even use property testing?

Reply:
property-based testing is usually better than unit testing where it works; imo unit testing should be mainly used for known corner cases